src/internal/mod.rs

```diff
@@ -828,21 +821,19 @@ pub(crate) mod test {
         ])
         .unwrap();
         let context = DeviceContext::new(
-            314.try_into()?,
             "account_id".try_into()?,
             22,
             priv_key.into(),
             DeviceSigningKeyPair::from(dev_keys),
         );
         let json = serde_json::to_string(&context).unwrap();
-        let expect_json = r#"{"deviceId":314,"accountId":"account_id","segmentId":22,"signingPrivateKey":"AQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQGKiOPddAnxlf1S2y08ul1yymcJvx2UEhvzdIgBtA9vXA==","devicePrivateKey":"bzb0Rlg0u7gx9wDuk1ppRI77OH/0ferXleenJ3Ag6Jg="}"#;
+        let expect_json = r#"{"accountId":"account_id","segmentId":22,"signingPrivateKey":"AQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQGKiOPddAnxlf1S2y08ul1yymcJvx2UEhvzdIgBtA9vXA==","devicePrivateKey":"bzb0Rlg0u7gx9wDuk1ppRI77OH/0ferXleenJ3Ag6Jg="}"#;
```

Comment:
Just curious, would this fail if the deviceId is in this json, or does serde just ignore it?

Reply:
This string is just used to check the DeviceContext serialization, so it needs to be removed here since it's no longer in the struct. But if I deserialized this string into a new DeviceContext, it would ignore the deviceId and work fine.